File: Src/java/engine-fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/terminology/HeaderInjectionInterceptor.java
/*
 *
 *   This code is derived from work done by Barry Cassidy from Motive Medical Intelligence
 *   Thank you!
 *
 * */

package org.opencds.cqf.cql.engine.fhir.terminology;

import ca.uhn.fhir.rest.client.api.IClientInterceptor;
import ca.uhn.fhir.rest.client.api.IHttpRequest;
import ca.uhn.fhir.rest.client.api.IHttpResponse;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class HeaderInjectionInterceptor implements IClientInterceptor {

    private HashMap<String, String> headers;

    /**
     * Instantiates a new header injection interception.
     *
     * @param headerKey the header key
     * @param headerValue the header value
     */
    public HeaderInjectionInterceptor(String headerKey, String headerValue) {
        super();
        this.headers = new HashMap<>();
        this.headers.put(headerKey, headerValue);
    }

    /**
     * Instantiates a new header injection interception.
     *
     * @param headers the headers
     */
    public HeaderInjectionInterceptor(HashMap<String, String> headers) {
        super();
        this.headers = headers;
    }

    @Override
    public void interceptRequest(IHttpRequest request) {

        for (Map.Entry<String, String> entry : this.headers.entrySet()) {
            request.addHeader(entry.getKey(), entry.getValue());
        }
    }

    @Override
    public void interceptResponse(IHttpResponse response) throws IOException {
        // nothing
    }
}

File: Src/java/engine-fhir/src/main/kotlin/org/opencds/cqf/cql/engine/fhir/terminology/HeaderInjectionInterceptor.kt
/*
 *
 *   This code is derived from work done by Barry Cassidy from Motive Medical Intelligence
 *   Thank you!
 *
 * */
package org.opencds.cqf.cql.engine.fhir.terminology

import ca.uhn.fhir.rest.client.api.IClientInterceptor
import ca.uhn.fhir.rest.client.api.IHttpRequest
import ca.uhn.fhir.rest.client.api.IHttpResponse
import java.io.IOException

class HeaderInjectionInterceptor(private val headers: Map<String, String> = mapOf()) :
    IClientInterceptor {
    /**
     * Instantiates a new header injection interception.
     *
     * @param headerKey the header key
     * @param headerValue the header value
     */
    constructor(headerKey: String, headerValue: String) : this(mapOf(headerKey to headerValue))

    override fun interceptRequest(request: IHttpRequest) {
        for (entry in this.headers.entries) {
            request.addHeader(entry.key, entry.value)
        }
    }

    @Throws(IOException::class)
    override fun interceptResponse(response: IHttpResponse?) {
        // nothing
    }
}
